### PR TITLE
added support for multiple currencies

### PR DESCRIPTION
**Pull request by rohitcbr**
### - Link added should be a valid URL and accessible throughout the org.
   ### - If the branch name contains hotfix / revert by default the BVT workflow check will pass.

| Test Case Document URL                        |
|-----------------------------------------------|
| Please paste test case document link here.... |

### CODE REVIEW

File: supported-currencies.json [hunk 1]
```diff
@@ -0,0 +1,544 @@
+{
+    "supported-currencies": [
+      {
+        "Currency Name": "United Arab Emirates Dirham",
```
Comment (abdulwahidsharief) [inline supported-currencies.json:4]: make Currency Name as currency_name

File: supported-currencies.json [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+        "ISO Code": "AED",
+        "Exponent": "2"
+      },
+      {
+        "Currency Name": "Albanian lek",
+        "ISO Code": "ALL",
```
Comment (abdulwahidsharief) [inline supported-currencies.json:10]: make ISO code as iso_code

File: supported-currencies.json [hunk 3]
```diff
@@ -0,0 +1,544 @@
...
+        "Exponent": "2"
+      },
+      {
+        "Currency Name": "Armenian dram",
+        "ISO Code": "AMD",
+        "Exponent": "2"
```
Comment (abdulwahidsharief) [inline supported-currencies.json:16]: use small case for keys

File: templates/razorpay-settings-templates.php [hunk 4]
```diff
@@ -156,6 +156,42 @@ function displayPaymentAction()
...
+        { 
+            if ($value === $default)
+            {
+                $currency .='<option value="' . $value . '" selected="selected"'; 
+            }                        
+            else
+            {
+                $currency .='<option value="' . $value . '"'; 
+            }
+            $currency .= '>' . $value . '</option>';
```
Comment (abdulwahidsharief) [inline templates/razorpay-settings-templates.php:187]: close option in one line

File: templates/razorpay-settings-templates.php [hunk 5]
```diff
@@ -156,6 +156,42 @@ function displayPaymentAction()
...
+        }
+        $currency .='</select>'; 
+        $currency .='<br><label>Please choose a currency from the drop-down list.</label>';
```
Comment (abdulwahidsharief) [inline templates/razorpay-settings-templates.php:190]: leave space after =

File: templates/razorpay-settings-templates.php [hunk 6]
```diff
@@ -156,6 +156,42 @@ function displayPaymentAction()
         echo $paymentAction;
     }
 
+        /**
```
[the remark this answers is not in the document]
Comment (rohitcbr) [reply templates/razorpay-settings-templates.php]: fixed here https://github.com/razorpay/razorpay-quick-payments/blob/b7757b7948021cd22c1eb9d5326e5ea8c3decb1b/templates/razorpay-settings-templates.php#L159

File: templates/razorpay-settings-templates.php [hunk 7]
```diff
@@ -156,6 +156,42 @@ function displayPaymentAction()
...
+     * Currency field of settings page
+    **/
+    function displayCurrencyAction()
+    {
+        $default = get_option('currency_action_field');
+
+        $supported_currencies = json_decode(file_get_contents(__DIR__."/../supported-currencies.json"), true)['supported-currencies'];
```
[the remark this answers is not in the document]
Comment (rohitcbr) [reply templates/razorpay-settings-templates.php]: fixed here https://github.com/razorpay/razorpay-quick-payments/blob/b7757b7948021cd22c1eb9d5326e5ea8c3decb1b/templates/razorpay-settings-templates.php#L166

File: templates/razorpay-settings-templates.php [hunk 8]
```diff
@@ -156,6 +156,42 @@ function displayPaymentAction()
...
+
+        $supported_currency_codes = [];
+
+        foreach ( $supported_currencies as $k => $v )
```
[the remark this answers is not in the document]
Comment (rohitcbr) [reply templates/razorpay-settings-templates.php]: fixed here https://github.com/razorpay/razorpay-quick-payments/blob/b7757b7948021cd22c1eb9d5326e5ea8c3decb1b/templates/razorpay-settings-templates.php#L170

File: supported-currencies.json (hunk 1 above)
Comment (rohitcbr) [reply supported-currencies.json]: fixed here https://github.com/razorpay/razorpay-quick-payments/blob/b7757b7948021cd22c1eb9d5326e5ea8c3decb1b/supported-currencies.json#L4

File: supported-currencies.json (hunk 2 above)
Comment (rohitcbr) [reply supported-currencies.json]: fixed here https://github.com/razorpay/razorpay-quick-payments/blob/b7757b7948021cd22c1eb9d5326e5ea8c3decb1b/supported-currencies.json#L5

File: supported-currencies.json (hunk 3 above)
Comment (rohitcbr) [reply supported-currencies.json]: fixed here https://github.com/razorpay/razorpay-quick-payments/blob/b7757b7948021cd22c1eb9d5326e5ea8c3decb1b/supported-currencies.json#L6

File: templates/razorpay-settings-templates.php (hunk 4 above)
Comment (rohitcbr) [reply templates/razorpay-settings-templates.php]: fixed here https://github.com/razorpay/razorpay-quick-payments/blob/b7757b7948021cd22c1eb9d5326e5ea8c3decb1b/templates/razorpay-settings-templates.php#L191

File: templates/razorpay-settings-templates.php (hunk 5 above)
Comment (rohitcbr) [reply templates/razorpay-settings-templates.php]: fixed here https://github.com/razorpay/razorpay-quick-payments/blob/b7757b7948021cd22c1eb9d5326e5ea8c3decb1b/templates/razorpay-settings-templates.php#L196

File: templates/razorpay-settings-templates.php [hunk 9]
```diff
@@ -156,6 +156,48 @@ function displayPaymentAction()
...
+                    
+        foreach ($supported_currency_codes as $index => $value)
+        { 
+            if ($value === $default)
+            {
+                $currency .= '<option value="' . $value . '" selected="selected">' . $value . '</option>'; 
+            }                        
+            else
+            {
+                if (empty($default) === true && $value === 'INR')
```
Comment (abdulwahidsharief) [inline templates/razorpay-settings-templates.php:185]: use and instead &&

**Review by abdulwahidsharief — APPROVED** [review]
LGTM

File: templates/razorpay-settings-templates.php (hunk 9 above)
Comment (rohitcbr) [reply templates/razorpay-settings-templates.php]: fixed here https://github.com/razorpay/razorpay-quick-payments/blob/4bf92b0b3e7bec74ab0bb3ee9f469fb902639578/templates/razorpay-settings-templates.php#L185

**Pull request merged**